client/web/src/repo/blob/backend.ts

```diff
@@ -3,76 +3,84 @@ import { map } from 'rxjs/operators'
 
 import { memoizeObservable } from '@sourcegraph/common'
 import { dataOrThrowErrors, gql } from '@sourcegraph/http-client'
-import { ParsedRepoURI, makeRepoURI } from '@sourcegraph/shared/src/util/url'
+import { makeRepoURI } from '@sourcegraph/shared/src/util/url'
 
 import { requestGraphQL } from '../../backend/graphql'
 import { BlobFileFields, BlobResult, BlobVariables, HighlightResponseFormat } from '../../graphql-operations'
 
-function fetchBlobCacheKey(parsed: ParsedRepoURI & { disableTimeout?: boolean; format?: string }): string {
-    return `${makeRepoURI(parsed)}?disableTimeout=${parsed.disableTimeout}&=${parsed.format}`
+/**
+ * Makes sure that default values are applied consistently for the cache key and the `fetchBlob` function.
+ */
+function applyDefaultValuesToFetchBlobOptions(options: FetchBlobOptions): Required<FetchBlobOptions> {
+    const { disableTimeout = false, format = HighlightResponseFormat.HTML_HIGHLIGHT } = options
+
+    return {
+        ...options,
+        disableTimeout,
+        format,
+    }
+}
```

Comment:
Minor suggestion:
```suggestion
const applyDefaultValuesToFetchBlobOptions = ({ disableTimeout = false, format = HighlightResponseFormat.HTML_HIGHLIGHT, ...options }: FetchBlobOptions): Required<FetchBlobOptions> => ({
    ...options,
    disableTimeout,
    format,
})
```
Or
```suggestion
const applyDefaultValuesToFetchBlobOptions = (options: FetchBlobOptions): Required<FetchBlobOptions> => ({
    ...options,
    disableTimeout: options.disableTimeout ?? false,
    format: options.format ?? HighlightResponseFormat.HTML_HIGHLIGHT,
})
```